src/miniflask/state.py

```diff
@@ -394,7 +374,156 @@ def __call__(self, state, event):  # pylint: disable=redefined-outer-name
     def str(self, asciicodes=True, color_attr=attr):
         if not asciicodes:
             color_attr = lambda x: ''  # noqa: E731 no-lambda
-        return color_attr('dim') + "'" + str(self.type) + "' or '" + "None" + "' ⟶   " + color_attr('reset') + str(self.default)
+        return color_attr('dim') + "'" + str(self.type) + "' or '" + "None" + "' ⟶   " + color_attr('reset') + str(self.dependencies)
+
+    def __str__(self):
+        return self.str()
+
+
+state_regex = re.compile(r"state\[(\"|\')((?:\.*\w+)+)\1\]")
+string_regex_g2 = r"([\"'])((?:\\\1|(?:(?!\1)).)*)(?:\1)"
+if_else_regex = re.compile(r"(.*)if\s+(.*)\s+else(.*)")
+state_in_regex = re.compile(string_regex_g2 + r"\s+in\s+state")
+
+
+class state_node:
+    def __init__(self, varid, mf, caller_traceback, cliargs=False, parsefn=False, is_ovewriting=False, missing_argument_message=None, fn=None):
+        self.varid = varid
+        self.mf = mf
+        self.caller_traceback = caller_traceback
+        self.cliargs = cliargs
+        self.parsefn = parsefn
+        self.is_ovewriting = is_ovewriting
+        self.cli_overwritten = False
+        self.missing_argument_message = missing_argument_message
+
+        self.depends_on = []
+        self.depends_alternatives = {}
+
+        self.fn = fn
+        self.fn_src = getsource(self.fn) if self.fn is not None else None
+
+        if self.fn_src is not None:
```

Comment:
Why is the code checked manually, instead of parsing the complete expression and creating the AST, looking for variable usage?

Reply:
If have implemented the dependency checking via the python AST, see [miniflask/dev-ast](https://github.com/da-h/miniflask/tree/dev-ast) branch.